fetch/mesh_data.go

```diff
@@ -48,18 +48,26 @@ func (f *Fetch) getHashes(ctx context.Context, hashes []types.Hash32, hint datas
 				return ctx.Err()
 			case <-p.completed:
 				if p.err != nil {
+					// do not log err here, as it can be a multierror underneath
 					f.logger.WithContext(ctx).With().Warning("failed to get hash",
 						log.String("hint", string(hint)),
 						log.Stringer("hash", h),
-						log.Err(p.err))
+					)
 					return p.err
 				}
 			}
 			return nil
 		})
 	}
 
-	return eg.Wait().ErrorOrNil()
+	err := eg.Wait().ErrorOrNil()
+	if err != nil {
+		// due to the recursive nature of data fetching in go-spacemesh, we want to minimize the log volume
+		f.logger.WithContext(ctx).With().Debug("failed to get hashes",
+			log.String("hint", string(hint)),
+			log.Err(err))
+	}
```

Comment:
Why log it at all here? IMO the code that handles the error should log it. Any contextual information like `hint` could be backed into the returned error.

Reply:
removed. thanks.